astropy/stats/funcs.py

```diff
@@ -1098,19 +1098,13 @@ def _scipy_kraft_burrows_nousek(N, B, CL):
 
     from scipy.optimize import brentq
     from scipy.integrate import quad
+    from scipy.special import factorial
 
     from math import exp
 
     def eqn8(N, B):
         n = np.arange(N + 1, dtype=np.float64)
```

Comment:
```suggestion
        n = np.arange(N + 1, dtype=int)
```

Shouldn't `n` be an integer?  A `factorial` of a float doesn't make sense to me?

Reply:
I didn't write that line... Should I change it anyway?

Reply:
The docstring for `scipy.special.factorial` says `n` should be an int or array of int:  https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.factorial.html.  I think we should try it, and see if any tests fail. :). We may want to also add `exact=True` in the `factorial` call if there are any small floating-point errors.

Reply:
> The docstring for `scipy.special.factorial` says `n` should be an int or array of int: 

The docs aren't entirely accurate; the `exact` version indeed only accepts integers, but with `exact=False` apparently simply returns `gamma(n+1)`.

>  We may want to also add `exact=True` in the `factorial` call if there are any small floating-point errors.

See my earlier comment; depends if any  n > 20 are expected; for larger values the return would need to be stored in `np.float64`.

Reply:
Sounds good.  @pllim, so leave it as you currently have it with n as float.

Reply:
Yes, from the docstring (and the application case) N at least up to 100 can be expected.

Reply:
I am wondering if the
`factorial_N = float(math.factorial(N))`
below could not be simply replaced by
`factorial_N = factorial(N)`
as well. It would solve #10832 along the way, at the cost of introducing some `rtol < 1.e-15` inaccuracies for large N.

Reply:
Isn't changing this beyond the scope of this PR?

Reply:
Okay, leaving as-is then. Thanks!